pylint/config/config_file_parser.py

```diff
@@ -42,6 +42,15 @@ def _parse_ini_file(file_path: Path) -> tuple[dict[str, str], list[str]]:
         config_content: dict[str, str] = {}
         options: list[str] = []
         for section in parser.sections():
+            if "setup.cfg" in str(file_path) and not section.startswith("pylint"):
```

Comment:
What if we use the ``--rcfile`` option and the file is not named "setup.cfg" ?

Reply:
Then we load every section like we do now. I chose to do this because if people are going to the trouble of defining an `ini` file and pointing to it it is very likely that this is indeed a `pylintrc` or other `pylint` specific file. I don't think we want to enforce `pylint.main` in `pylintrc` itself.

Reply:
> I don't think we want to enforce pylint.main in pylintrc itself.

Yeah not unless we have a migration tool ready.

Reply:
Hmm wait, how do we know that this section was supposed to be related to pylint and not flake8 ?

Reply:
It is not going to raise a warning for an empty `setup.cfg` but it is for any `[MASTER]` or `[master]` section in a file called `setup.cfg`. I think that is the only way to even raise a warning.

Reply:
That would be nice, but the test is ``and not section.startswith("pylint"):`` not ``"master" not in section.lower()`` ?

Reply:
We `continue` for `else`? We emit the warning for `master` and then continue parsing it. For all other sections that don't start with `pylint` we simply `continue`.

So `[MESSAGES CONTROL]` and `[flake8.Master]` won't be read any more.

Reply:
Ha right I missed that, my bad.